kong/init.lua

```diff
@@ -98,9 +99,95 @@ local ffi = require "ffi"
 local cast = ffi.cast
 local voidpp = ffi.typeof("void**")
 
+local plugins_map_version
+local plugins_map_semaphore
+
+local PLUGINS_MAP_CACHE_OPTS = { ttl = 0 }
+
+local configured_plugins
 local loaded_plugins
 local schema_state
 
+local function build_plugins_map(db, version)
+  local map = {}
+
+  for plugin, err in db.plugins:each() do
+    if err then
+      return nil, tostring(err)
+    end
+
+    map[plugin.name] = true
+  end
+
+  if version then
+    plugins_map_version = version
+  end
+
+  configured_plugins = map
+
+  return true
+end
+
+local function plugins_map_wrapper()
```

Comment:
The design of this function is slightly unaligned with our current best practices. 

Instead of logging and returning `false`, it'd be preferable if it were returning `nil, err`.
This way, the caller decides what to do of the error message, and in this particular case, at this particular time, we want to log it with a `CRIT` level, indeed.

So in the current form of this patch, the `rewrite` phase call would be written as:
```lua
local ok, err = plugins_map_wrapper()
if not ok then
  ngx.log(ngx.CRIT, "could not ensure plugins map is up to date: ", err)
  return HTTP_500
end
```

(This came from a weird feeling after reading the `local ok =  build_map(); if not ok then` branch of the `rewrite` phase, and I think this concern explains why this original feeling).

And this function returns one of those:

* `return nil, err`
* `"return nil, failed to acquire plugins map mutex"`

Reply:
adjusted